src/scopeHandler.js

```diff
@@ -25,6 +25,8 @@ const alwaysThrowHandler = new Proxy(freeze({}), {
  * - route all other property lookups at the safeGlobal.
  * - hide the unsafeGlobal which lives on the scope chain above the 'with'.
  * - ensure the Proxy invariants despite some global properties being frozen.
+ *
+ * @returns {ProxyHandler<any> & Record<string, any>}
```

Comment:
I never did learn this comment-type language, so I can't review whether this declaration is good. Since it is in a comment, I won't block a merge here on my own ignorance.

Reply:
It's using jsdoc-like syntax to hint ts server about the type of the function